packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -864,6 +879,23 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
     return new Import(scope, id);
   }
 
+  /**
+   * Imports an existing autoscaling group.
+   */
+  public static fromAutoScalingGroupAttributes(scope: Construct, id: string, attrs: AutoScalingGroupAttributes): IAutoScalingGroup {
```

Comment:
There is a lot of duplication between this method and `fromAutoScalingGroupName()`. Could you refactor this so that `fromAutoScalingGroupName()` calls `fromAutoScalingGroupAttributes()`, similarly to how `fromBucketName()` calls `fromBucketAttributes()` [here](https://github.com/aws/aws-cdk/blob/3e9f04dbbd7aadb8ab4394fefd6281f1d6d30fe0/packages/%40aws-cdk/aws-s3/lib/bucket.ts#L1368)?